src/realm/table.cpp

```diff
@@ -1053,12 +1053,20 @@ bool Table::set_embedded(bool embedded)
         });
 
         if (has_backlink_columns) {
-            for (auto o : *this) {
-                // each object should be owned by one and only one parent
-                if (o.get_backlink_count() != 1) {
+            for (auto object : *this) {
+                if (object.get_backlink_count() == 0) {
+                    // If the object does not have any backlinks it can be deleted.
+                    object.remove();
+                } else if (object.get_backlink_count() > 1) {
+                    // Each object should be owned by one parent at most.
                     return false;
                 }
             }
+        } else {
+            // If the table does not have any backlink columns the objects can be deleted.
+            for (auto object : *this) {
```

Comment:
If the table has no backlink columns, then the objects cannot be owned by objects in another table. So the table cannot be embedded.
Anyway - I would prefer `clear()` here.

Reply:
Having an embedded type with no incoming links will cause a schema validation error in AdditiveExplicit mode (since https://github.com/realm/realm-core/pull/4306). There is clearly some error here but I'm not sure if we should allow it here and fail later or what. To me it seems likely that the developer made a mistake, and if the change to embedded was an accident or test, it might be unwelcome to have all the data erased. For that reason, I suggest throwing an error here. Perhaps the error message can tell the user to first make a link to the table before changing it to be embedded.